### ISSUE TITLE
Remix v1.16.0 - Unnecessary Node polyfills added for Oxygen/Workers.

### ISSUE THREAD

**Issue by frandiox** [remix-run/remix#6348]
### What version of Remix are you using?

1.16.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

I think this was introduced in https://github.com/remix-run/remix/pull/5274 , and is probably related to https://github.com/remix-run/remix/issues/6280 and its linked issues/PRs. I'm still writing this issue to show that it also affects Worker environments

You can reproduce the issue when building a project with a typical config for Oxygen or CFW:

```js
module.exports = {
  appDirectory: 'app',
  ignoredRouteFiles: ['**/.*'],
  server: './server.ts',
  serverMainFields: ['browser', 'module', 'main'],
  serverConditions: ['worker', process.env.NODE_ENV],
  serverDependenciesToBundle: 'all',
  serverModuleFormat: 'esm',
  serverPlatform: 'neutral',
  serverMinify: false, // To check the bundle later
};
```

### Expected Behavior

The resulting server/worker bundle shouldn't contain Node polyfills. Or, at least, they shouldn't make the bundle crash when running on a Worker environment.

### Actual Behavior

Many Node polyfills are added to the JS and Server bundles, thus increasing the size by ~60K non-minified. One of the polyfills is also accessing a global `navigator` variable, making the process exit with `navigator is not defined` in Worker environments:

<img width="714" alt="image" src="https://user-images.githubusercontent.com/1634092/236779874-d5e70b4b-4bc9-441e-b2fa-de8851f31b95.png">

<img width="984" alt="image" src="https://user-images.githubusercontent.com/1634092/236782124-a882e286-c7a2-4381-93aa-c4c8956dc0f4.png">

---

I'm not sure if the solution is modifying [this conditional](https://github.com/remix-run/remix/blob/bac8b5bf9c9c483a6aaed2db813782a546e40148/packages/remix-dev/compiler/server/compiler.ts#L63-L65) and skip when `serverPlatform === 'neutral'`, or perhaps passing different options to [esbuild-plugin-polyfill-node](https://www.npmjs.com/package/esbuild-plugin-polyfill-node) since it seems to have a different default behavior compared to the old [@esbuild-plugins/node-modules-polyfill](https://www.npmjs.com/package/@esbuild-plugins/node-modules-polyfill).

**Comment by MichaelDeBoey** [comment]
We just published version `1.17.1-pre.0` which includes #6562 (a possible fix). If you'd like to take it for a test run please try it out and let us know what you think!

**Comment by frandiox** [comment]
Thanks! I can confirm that #6562 fixes this issue. Should we close it now or wait until 1.17.1 is released?

**Comment by MichaelDeBoey** [comment]
Closed by #6562
